tokens/src/lib.rs

```diff
@@ -890,6 +901,7 @@ impl<T: Config> Pallet<T> {
 		amount: T::Balance,
 		existence_requirement: ExistenceRequirement,
 	) -> DispatchResult {
+		T::OnTransfer::on_transfer(currency_id, from, to, amount)?;
```

Comment:
move this after the early return. on the case no event is emitted, no hook should be called.